### PR TITLE
Reduce ClientSecrets scenario load

### LINKED ISSUE #1018: Why 20k clients is so slow
Cache usage: 2x in users cache, 3x in realms cache per client. 

-> 10k clients

-> reduce RPS to a number where the DB is no longer overload -> reduce to 50% to not overload the database

-> Write some docs on how to size users and client cache based on the number of clients

Follow-up task: update the sizing guide for RHBK26

### PR DESCRIPTION
**Pull request by pruivo**
* Current performance run is (over)loading the DB too much (98% CPU)
* Reduce the total number of clients to 10k (75% cache hit ratio)
* Reduce the requests per second to 750 (25% reduction)

Closes #1018

### CODE REVIEW

File: .github/workflows/rosa-scaling-benchmark.yml [hunk 1]
```diff
@@ -94,6 +94,7 @@ env:
   PROJECT: runner-keycloak
   ANSIBLE_CUSTOM_VARS_ARG: '-e @env_rosa_benchmark.yml'
   CLIENTS_PER_REALM: 20000
+  GATLING_CLIENTS_PER_REALM: 10000
```
Comment (mhajas) [inline .github/workflows/rosa-scaling-benchmark.yml:97]: Why are we distinguishing between clients created and clients used in load running? Should we just change the variable CLIENTS_PER_REALM to 10k?

File: provision/rosa-cross-dc/Taskfile.yaml [hunk 2]
```diff
@@ -792,6 +792,7 @@ tasks:
     vars:
       SCENARIO: '{{.SCENARIO | default "keycloak.scenario.authentication.AuthorizationCode"}}'
       USERS: "{{.USERS | default 20000}}"
+      CLIENTS: "{{.CLIENTS | default 10000}}"
```
Comment (mhajas) [inline provision/rosa-cross-dc/Taskfile.yaml:795]: Is this used somewhere? Should we remove from the Taskfile if not?

File: .github/workflows/rosa-scaling-benchmark.yml [hunk 3]
```diff
@@ -378,7 +379,7 @@ jobs:
           --ramp-up=20
           --measurement=${{ inputs.measurement }}
           --users-per-realm=${{ inputs.usersPerRealm }}
-          --clients-per-realm=${{ env.CLIENTS_PER_REALM }}
+          --clients-per-realm=${{ env.GATLING_CLIENTS_PER_REALM }}
```
Comment (mhajas) [inline .github/workflows/rosa-scaling-benchmark.yml:382]: Should we change this for all test runs not only for the ClientSecret scenario?

File: .github/workflows/rosa-scaling-benchmark.yml (hunk 1 above)
Comment (pruivo) [reply .github/workflows/rosa-scaling-benchmark.yml:97]: I though we still wanted to create 20k but only use 10k. If I'm wrong, then I can remove this variable.

File: provision/rosa-cross-dc/Taskfile.yaml (hunk 2 above)
Comment (pruivo) [reply provision/rosa-cross-dc/Taskfile.yaml:795]: I use the task file to trigger the runs when I need it. I want to keep it in sync with the nightly defaults.

File: .github/workflows/rosa-scaling-benchmark.yml (hunk 1 above)
Comment (mhajas) [reply .github/workflows/rosa-scaling-benchmark.yml:97]: I don't think there is much difference between creating 20k and 10k so I would just use 10k everywhere to save time creating the additional 10k. Also, we previously used 1 client-per-realm everywhere, so 10k is still better.

**Pull request merged**